Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -685,10 +698,41 @@ bool Renderer::IsFrameDumping()
   if (SConfig::GetInstance().m_DumpFrames)
     return true;
 
-  ShutdownFrameDumping();
   return false;
 }
 
+void Renderer::DoDumpFrame()
+{
+  UpdateFrameDumpTexture();
+
+  auto result = m_dump_texture->Map();
```

Comment:
Has there been any performance comparisons with frame dumping? I suspect this will be slower than the current mechanism, particularly in Vulkan and GL. Frame dumping already is a performance hit, we probably don't want to make it slower :/

Reply:
No, I didn't do any performance tests.  Early on, performance wasn't really a concern, just getting everything working.  I can look into this but frame dumping has never really been performant.  I don't know how much time we want to work on this vs doing it in a separate PR, so that we don't block the merge.